javascript/require-and-import-mini-lesson.md

```diff
@@ -0,0 +1,351 @@
+
+|REQUIRE|IMPORT|
+|--|--|
+|Can be called at any time and place in the program.|Always run in the beginning of the file.|
+|You can directly run the code with the require statement.|If you are running Node 13 and above, to run a program containing import statement you have to use the `--experimental-modules` flag.|
+|If you want to use the require module, you have to save file with ‘.js’ extension.|If you are running Node 13 and above, if you want to use import module, you have to save file with ‘.mjs’ extension.|
+|Can leave out a .js extension when importing a local module|Cannot leave the extension|
+|Loading is synchronous, which means that if you have multiple `require`, they are loaded and processed one by one.|Import is asynchronous.|
+|Loading is synchronous, which means that if you have got multiple requires, they’re loaded and processed one by one.| Import is asynchronous|
+|To include multiple functions, you first export the functions in an object, then import them using `require`|To include multiple functions, simply import them between curly braces.|
+
+
+### Guided Practice
+
+In order to be able to follow this guided practice, make sure you are using Node.js 13 and above. To install Node, simply go to [https://nodejs.org/en/download/](https://nodejs.org/en/download/) and choose the correct installation for your system.
+
+#### Using `require`
+
+The first thing we need to to is to create our export file. So, opening your favorite code editor, navigate to a new a folder for this practice and create a file named `my-module-for-require.js`. Inside the file, enter the following snippet:
+
+```javascript
+const messageToExport = "This message has been exported.";
+
+exports.exportedMessage = messageToExport;
+```
+
+This basically creates something to be exported and then it attaches it to the `exports` object.
+
+Then, you need to create the file in which we will call our module. So within the same folder for now, create a file named `example-for-require.js`, and then use `require` as such:
+
+```javascript
+// require your module
+const obj = require("./module-with-functions");
+// place the exported data in a constant
+const message = obj.exportedMessage;
+
+// console log it
+console.log(message);
+```
+
+Now, open a terminal, make sure you are inside of the folder where the `example-for-require.js` file is and enter the following command to run your program:
+
+```bash
+node example-for-require.js
+```
+
+If all goes well, you should see:
+
+```bash
+This message has been exported.
+```
+
+Awesome! You've just created a small module and used it in your own program. I'm sure you can appreciate all the possibilities!
+
+#### Using `import`
+
+As such, it's not possible to use `import` in Node directly. Now, as stated above, there are two ways that we can circumvent this. First, by using an `.msj` extension and second by creating a simple `package.json` file to set the environment. Let's try an example of each.
```

Comment:
Please remove the mjs practice as it's not something they will have to do and trying to use it for real has even frustrated advanced users, like me 😬

Having additional practice on using destructuring to directly require just the function above would be helpful to add, e.g.

```js
 const { message } = require("./module-with-functions");
```

Reply:
[c89273b](https://github.com/Techtonica/curriculum/pull/1414/commits/c89273b844a10c1f67a1d0886a773f18a1c327e8)